File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
      int arr[]=new int[nums.length];//to store result array
        int j=0;//to point even position 
        int k=1;//to point odd position
       for(int i=0;i<nums.length;i++)
       {
           if(nums[i]>0)//this if statement add positive number at even position
           {
               arr[j]=nums[i];
               j+=2; //j=0,2,4,6,8
           }
           else{
               arr[k]=nums[i];//add negetive number at odd position
               k+=2;//k=1,3,5,7,9
           }
       }
       return arr; //return array 
        
    }
}

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.py
class Solution:
    def rearrangeArray(self, nums: List[int]) -> List[int]:
        pos = []
        neg = []

        for num in nums:
            if num>0:
                pos.append(num)
            elif num <= 0:
                neg.append(num)
        n = len(pos)
        res = []

        for i in range(0,n):
            res.append(pos[i])
            res.append(neg[i])

        return res



        